Jenkinsfile-downloads

```diff
@@ -27,25 +27,21 @@ pipeline{
 		stage('Setup: Build jar file'){
 			steps{
 				script{
-					dir('add-links'){
-						sh "mvn clean package -DskipTests"
-					}
+					sh "mvn clean package -DskipTests"
 				}
 			}
 		}
 		// This stage executes the AddLinks file download step. It will download all files needed to run the AddLinks insertion step, storing them
 		// at /tmp/addlinks-downloaded-files.
 		stage('Main: Download AddLinks files'){
 			steps{
-				script{
-					dir('add-links'){
-						withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
-							sh "cp $ConfigFile src/main/resources/auth.properties"
-							sh "ln -sf src/main/resources resources"
-							sh "rm -f src/main/resources/db.properties"
-							sh """java -cp "\$(pwd)/resources" -Dconfig.location=\$(pwd)/resources/addlinks.properties -Dlog4j.configurationFile=\$(pwd)/resources/log4j2.xml -Xmx${env.JAVA_MEM_MAX}m -jar target/AddLinks-${env.ADDLINKS_VERSION}.jar file://\$(pwd)/resources/file-download-context.xml"""
-							sh "rm -f src/main/resources/auth.properties"
-						}
+			script{
+					withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
+						sh "cp $ConfigFile src/main/resources/auth.properties"
+						sh "ln -sf src/main/resources resources"
+						sh "rm -f src/main/resources/db.properties"
+						sh """java -cp "\$(pwd)/resources" -Dconfig.location=\$(pwd)/resources/addlinks.properties -Dlog4j.configurationFile=\$(pwd)/resources/log4j2.xml -Xmx${env.JAVA_MEM_MAX}m -jar target/AddLinks-*-SHADED.jar file://\$(pwd)/resources/file-download-context.xml"""
```

Comment:
Just curious what the triple quote does here?

Reply:
Its the same as normal quotes, but it allows the $(pwd) that is called within the command, and itself is surrounded by normal quotes, to be executed without issue. I found when using the normal quotes with this particular command that that it wasn't executed properly by Jenkins, because it would think that 'java -cp' was the command and would fail. The triple quotes was how I got around it

Reply:
Got it - thanks!